quickwit/quickwit-serve/src/otlp_api/rest_handler.rs

```diff
@@ -0,0 +1,255 @@
+// Copyright (C) 2023 Quickwit, Inc.
+//
+// Quickwit is offered under the AGPL v3.0 and as commercial software.
+// For commercial licensing, contact us at hello@quickwit.io.
+//
+// AGPL:
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use bytes::Bytes;
+use quickwit_opentelemetry::otlp::{OtlpGrpcLogsService, OtlpGrpcTracesService};
+use quickwit_proto::opentelemetry::proto::collector::logs::v1::logs_service_server::LogsService;
+use quickwit_proto::opentelemetry::proto::collector::logs::v1::{
+    ExportLogsServiceRequest, ExportLogsServiceResponse,
+};
+use quickwit_proto::opentelemetry::proto::collector::trace::v1::trace_service_server::TraceService;
+use quickwit_proto::opentelemetry::proto::collector::trace::v1::{
+    ExportTraceServiceRequest, ExportTraceServiceResponse,
+};
+use quickwit_proto::{tonic, ServiceError, ServiceErrorCode};
+use serde::{self, Serialize};
+use tracing::error;
+use warp::{Filter, Rejection};
+
+use crate::json_api_response::make_json_api_response;
+use crate::{require, with_arg, BodyFormat};
+
+#[derive(utoipa::OpenApi)]
+#[openapi(paths())]
+pub(crate) struct OtlpApi;
+
+/// Setup OpenTelemetry API handlers.
```

Comment:
Should we make an update in `openapi.rs`?

Reply:
Yes I did not implement it here. Do you want to add it in a follow up PR?

Reply:
yeah, happy to do that later today